### DISCUSSION TITLE
Very slow APIs, many times over 20 seconds latency

### DISCUSSION THREAD

**Discussion by nishuth** [deepgram/community#751] [General help] [answered]
I'm facing this repeated issues with API response time. Today it's really bad, my last few request took more than 20 seconds time and once even more time.

Since there is no support available to report this kind of issues in production, I have to rely to report on this community where users can't really help.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by nishuth** [reply]
this is general latency issues we have been facing a lot recently, however status page eon your server says everything is working fine.

**Reply by nishuth** [reply]
you guys keep asking for things I can not provide and issues are hanging without no care for your customers. can you not provide at least email support and look into this ?

**Comment by jpvajda** [comment]
@nishuth  Sorry you are having issues with Latency, there is a lot that can be at play here, can you provide more details about what exactly it is you are trying to do?

Additional info helps us troubleshoot:

* The request_id 
* The features you used or the full api.deepgram.com URL you sent your request to, including parameters.
*  Any code snippets you can share.

**Reply by nishuth** [reply]
live users are complaining about the performance, we don't have request ids. this was happening to me as well when I was testing in test flight. 


I'm not sure the issue is that complex, we are just simply doing STT and sentence is 8 words long. requests are made directly from App so can't share anything about this. but latency has been a major issue right now. and 20 seconds plus time for 8 words is very bad. 

how can the company help mitigate this ? because they don't offer any support.

**Reply by nishuth** [reply]
is there any other way you can look into my account, there aren't that may requests anyways. but only 8 words are taking 10-20 seconds is really bad performance.,

**Comment by jpvajda** [comment]
@nishuth  We'd actually need some type of information from you to track down your particular requests in our system. So the `Request ID` which is present in the metadata object of the API response from our API is useful for that purpose.

If you can share those we can look at specific requests that are having latency. 

Alternatively if you are willing to share a `project id` we can look up some information about your transactions this way.


## Where you can find the request ID

```
  "metadata": {
        "transaction_key": "deprecated",
        "request_id": "65567b9e-2a9b-49d2-b669-bd0131539ce8",     <------- HERE
        "sha256": "4dc60b6bdadb25060888f4df184d0a6db9cc217dcae748b2b4da9bde036cc062",
        "created": "2024-05-18T00:00:02.689Z",
        "duration": 17.193375,
        "channels": 1,
        "models": [
            "aa274f3c-e8b3-456a-ac08-dfd797d45514"
        ],
        "model_info": {
            "aa274f3c-e8b3-456a-ac08-dfd797d45514": {
                "name": "general-nova",
                "version": "2023-07-06.22746",
                "arch": "nova"
            }
        }
    },
```

## Support 

We do have support for our paid contract users which is something that can be added to your contract if you have one with us. 

We do offer community assistance for free , growth and pay as you go users.  If you this is the type of account you have we do our best to cover as many community issues and questions as we can.  You can read more about this in our [Community Code of Conduct](https://developers.deepgram.com/page/code-of-conduct#csa-community-service-agreement)

## Docs

Latency can be very difficult to troubleshoot and there are lot of factors at play. Depending on what you are doing here are some Docs that might help you

* https://developers.deepgram.com/docs/measuring-streaming-latency
* https://developers.deepgram.com/docs/troubleshooting-websocket-data-and-net-errors-when-live-streaming-audio

**Comment by nishuth** [comment]
I understand. But I’m trying to convey that this happens in production and at random how do I store IDs ?

I’m saying, performance degradation is something you should be monitoring. I’m sending just 8-10 words in audio file and it takes 20 seconds and sometimes 30 seconds at random.  But your website says your latency is better than any other provider.

Without ID isn’t there any other way for your team to look into performance degradation? This must be happening with others as well but their users may not be sharing, we are in MVP phase so we collect regular feedback but mature applications may not be tracking feedback like this. So investing time to solving this issue is good for your company as well.

Best,
Nish Patel
Sent from my iPhone

On May 17, 2024, at 5:10 PM, John Vajda (JV) ***@***.***> wrote:

﻿

@nishuth<https://github.com/nishuth> We'd actually need some type of information from you to track down your particular requests in our system. So the Request ID which is present in the metadata object of the API response from our API is useful for that purpose.

If you can share those we can look at specific requests that are having latency.

Alternatively if you are willing to share a project id we can look up some information about your transactions this way.

Where you can find the request ID

  "metadata": {
        "transaction_key": "deprecated",
        "request_id": "65567b9e-2a9b-49d2-b669-bd0131539ce8",     <------- HERE
        "sha256": "4dc60b6bdadb25060888f4df184d0a6db9cc217dcae748b2b4da9bde036cc062",
        "created": "2024-05-18T00:00:02.689Z",
        "duration": 17.193375,
        "channels": 1,
        "models": [
            "aa274f3c-e8b3-456a-ac08-dfd797d45514"
        ],
        "model_info": {
            "aa274f3c-e8b3-456a-ac08-dfd797d45514": {
                "name": "general-nova",
                "version": "2023-07-06.22746",
                "arch": "nova"
            }
        }
    },


Support

We do have support for our paid contract users which is something that can be added to your contract if you have one with us.

We do offer community assistance for free , growth and pay as you go users. If you this is the type of account you have we do our best to cover as many community issues and questions as we can. You can read more about this in our Community Code of Conduct<https://developers.deepgram.com/page/code-of-conduct#csa-community-service-agreement>

Docs

Latency can be very difficult to troubleshoot and there are lot of factors at play. Depending on what you are doing here are some Docs that might help you

  *   https://developers.deepgram.com/docs/measuring-streaming-latency
  *   https://developers.deepgram.com/docs/troubleshooting-websocket-data-and-net-errors-when-live-streaming-audio

**Comment by jpvajda** [comment]
@nishuth We do track system level performance, but what you are describing isn't being reported by others this is why we ask for something specific such as request IDs as it sounds like an isolated incident specific to what you are attempting to do. 

For reference our overall system performance is reported here: https://status.deepgram.com/ which you can always check for system wide outages or performance issues. 

Without specific Request ID or a specific Project ID (which is available after logging into console.deepgram.com)  there isn't much else I can do for your issue.

**Comment by jpvajda** [comment]
@nishuth  I'm curious are you based in India by any chance?

**Comment by nishuth** [comment, marked as answer]
Not. I’m in US. But some of the users are in India. My account is under same email account. ***@***.***

Also, speed has improved greatly since I have submitted this issue.

Best,
Nish Patel
Sent from my iPhone

On May 21, 2024, at 11:17 AM, John Vajda (JV) ***@***.***> wrote:

﻿

@nishuth<https://github.com/nishuth> I'm curious are you based in India by any chance?

**Comment by nishuth** [comment]
Hi,

My account is under Uth Ai. We are once again facing issues of long wait time.

Today it happened during my demo. 20 seconds plus for 8 words STT.

As I said this happens during live demos and user session so I’m not sure how I can send you IDs etc. but this MUST be resolved or I’ll have to leave the platform.

Best,
Nish Patel
Sent from my iPhone

On May 21, 2024, at 11:17 AM, John Vajda (JV) ***@***.***> wrote:

﻿

@nishuth<https://github.com/nishuth> I'm curious are you based in India by any chance?

**Comment by bd-g** [comment]
@nishuth 
> As I said this happens during live demos and user session so I’m not sure how I can send you IDs etc.

You can build in telemetry to collect system information from your user sessions. For example, collecting the request ID from a specific client-side user and recording it in a database or logging tool on your end associated with the user email. Then, if you get reports of increased latency, you can see all conversations associated with that user.

Telemetry is an important part of a production application. When looking at Deepgram's performance overall, there is not evidence of widespread degradation, which is why we need specific examples from you to investigate whether this is a specific issue scoped to your network or your code. 

If you have not implemented telemetry and do not provide a Project ID, we can't associate your GitHub user with any requests from your project on our end, and so can't investigate if there is something specific to your project that could be fixed. 

We are happy to assist you in debugging, but please be respectful and follow our [Code of Conduct](https://github.com/deepgram/deepgram-python-sdk/blob/main/.github/CODE_OF_CONDUCT.md) when seeking assistance.

**Comment by nishuth** [comment]
project ID= 21fc5476-a528-4661-9370-f333f24b7ccc

Not sure telemetry we can implement right now for various reasons.

Best,
Nish

On Jun 3, 2024, at 6:03 AM, Brent George ***@***.***> wrote:

﻿

@nishuth<https://github.com/nishuth>

As I said this happens during live demos and user session so I’m not sure how I can send you IDs etc.

You can build in telemetry to collect system information from your user sessions. For example, collecting the request ID from a specific client-side user and recording it in a database or logging tool on your end associated with the user email. Then, if you get reports of increased latency, you can see all conversations associated with that user.

Telemetry is an important part of a production application. When looking at Deepgram's performance overall, there is not evidence of widespread degradation, which is why we need specific examples from you to investigate whether this is a specific issue scoped to your network or your code.

If you have not implemented telemetry and do not provide a Project ID, we can't associate your GitHub user with any requests from your project on our end, and so can't investigate if there is something specific to your project that could be fixed.

We are happy to assist you in debugging, but please be respectful and follow our Code of Conduct<https://github.com/deepgram/deepgram-python-sdk/blob/main/.github/CODE_OF_CONDUCT.md> when seeking assistance.

**Comment by bd-g** [comment]
After reviewing your project, I can see you're using our pre-recorded services. Our pre-recorded speech-to-text offering is extremely fast, but may see elevated response times above 20 seconds during periods of high load.

This pre-recorded endpoint is designed to be fast, but it is not expected to be real time or have a maximum latency below 20 seconds. If you need consistently low latency times, please utilize our [Streaming speech to text services](https://developers.deepgram.com/docs/getting-started-with-live-streaming-audio), which process audio in real time and may be better suited for your use case.

**Comment by nishuth** [comment]
Thanks. Streaming audio doesn’t suite our needs and we expect more issues with it with deepgram because simple endpoint API has 20 seconds response time. Not to mention the price difference doesn’t make it affordable in our use case.

With all due respect, I’m Being technical I really don’t understand why an endpoint can’t be managed to provide good uptime and speed especially it has lower demand than LIVE streaming.

Best,
Nish Patel
Sent from my iPhone

On Jun 12, 2024, at 8:53 AM, Brent George ***@***.***> wrote:

﻿

After reviewing your project, I can see you're using our pre-recorded services. Our pre-recorded speech-to-text offering is extremely fast, but may see elevated response times above 20 seconds during periods of high load.

This pre-recorded endpoint is designed to be fast, but it is not expected to be real time or have a maximum latency below 20 seconds. If you need consistently low latency times, please utilize our Streaming speech to text services<https://developers.deepgram.com/docs/getting-started-with-live-streaming-audio>, which process audio in real time and may be better suited for your use case.

**Comment by bd-g** [comment]
For 8-10 second audio, this will almost always be processed extremely quickly. Waits of 20-30 seconds or longer are unexpected. At this time, we do not see performance degradation as you've described across our other users with similar audio profiles (short audio files).

As we described above, we will need specific request IDs to identify what appears to be an isolated incident specific to what you are attempting to do, or specific to your network.

**Comment by nishuth** [comment]
It’s was never resolved. Instead of solving issue I was asked to buy more expensive plan and reengineer my app to enable streaming audio STT.

Your ALIs can’t handle 8 words(2-3) seconds audio and take 20 seconds. Instead of fixing it I was asking to spend more money. No company has 20 seconds response time for 3 seconds audio.

I’m very disappointed.

Best,
Nish Patel
Sent from my iPhone

On Jun 22, 2024, at 9:03 AM, John Vajda (JV) ***@***.***> wrote:

﻿

Closed #751<https://github.com/orgs/deepgram/discussions/751> as resolved.

**Comment by bd-g** [comment]
Hello Nish,

We requested that you provide request IDs to track down issues with your specific project, as we are not seeing this behavior among other users. By providing a request ID, that would enable us to debug a specific network session and track down whether there was an issue with the API, or whether we need to take a second look at your own local network or application code. 

By using community support, you are asking to partner with us to work towards a solution. We provided next steps to iterate to a solution, including logging request IDs in your application. This is a basic instruction that should be implemented in any production application, but you suggested this was not possible. If you report a session is having latency issues, but cannot provide any details about that session, you are not giving us the information we need as we try to be good development partners. 

We are happy to continue to look into this further if you are able to provide the basic details requested, such as Deepgram Request ID where you observe the increased latency.

**Comment by SaidAbdullahi** [comment]
This is happening for me as well. Initially i was sending 2-3min audio and it ranged from 3s to 3mins. To mitigate I rearchitectured my app to do audio chunking, so now I am sending 10s chunks of the audio to the prerecorded api and it's still erratic, sometimes taking 1s and upto 20s. I believe the live streaming api should help as suggested but that is an even bigger change like Nish alluded to. I wonder what can be done to ensure consistently low latencies while using the prerecorded api? 
An example request id's:
838451eb-f528-49e4-ad93-4eff941b6761 this is a 10s audio that took 5.2s to transcribe according to my profiler

**Comment by nishuth** [comment]
Sorry to hear this, I still have no resolution for this. However, I’m planning replacing this API. I can’t rely on such unpredictable service, with no support.

In my opinion their tech may not be production ready. So many new AI companies have big claims right now but their product may not be production ready.

Lesson learned !

On Jul 31, 2024, at 2:06 AM, Said ***@***.***> wrote:

﻿

This is happening for me as well. Initially i was sending 2-3min audio and it ranged from 3s to 3mins. To mitigate I rearchitectured my app to do audio chunking, so now I am sending 10s chunks of the audio to the prerecorded api and it's still erratic, sometimes taking 1s and upto 20s. I believe the live streaming api should help as suggested but that is an even bigger change like Nish alluded to. I wonder what can be done to ensure consistently low latencies while using the prerecorded api?
An example request id's:
838451eb-f528-49e4-ad93-4eff941b6761 this is a 10s audio that took 5.2s to transcribe according to my profiler
